### PR TITLE
Charities and pages as collections

### PR DESCRIPTION
**Pull request by bradparker**

### CODE REVIEW

File: source/Routes/Home/index.js [hunk 1]
```diff
@@ -1,48 +1,98 @@
...
+        ...hero,
+        totalRaised,
+        totalTeams,
+        daysLeft,
+        totalMiles
+      }}
       about={about}
       route={route}
-      charities={{ charities }}
+      charities={{ charities: charities }}
```
Comment (alexander-bobin) [inline source/Routes/Home/index.js]: Could this be shorthand?

File: source/store/actions/charities/__tests__/fetchCharities-test.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    it('dispatches an action of type RECEIVE_CHARITIES_FAILURE', (done) => {
+      const { fetchCharities, dispatchSpy } = setup(() => Promise.reject())
+
+      fetchCharities(dispatchSpy)().catch(() => {
+        expect(dispatchSpy.getCall(1).args[0].type).to.eq('RECEIVE_CHARITIES_FAILURE')
+        done()
+      }).catch(done)
+    })
+
+    it('dispatches an action of type RECEIVE_CHARITIES_FAILURE', (done) => {
```
Comment (alexander-bobin) [inline source/store/actions/charities/__tests__/fetchCharities-test.js]: This test looks like be a mirror of the on prior

File: source/store/actions/charities/__tests__/fetchCharities-test.js [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+        done()
+      }).catch(done)
+    })
+  })
+
+  describe('when the fetch succeeds', () => {
+    it('dispatches an action of type RECEIVE_CHARITIES_SUCCESS', (done) => {
+      const { fetchCharities, dispatchSpy } = setup(() => Promise.resolve([]))
+
+      fetchCharities(dispatchSpy)().then(() => {
```
Comment (alexander-bobin) [inline source/store/actions/charities/__tests__/fetchCharities-test.js]: Not important but another way (may be less explicit) to handle tests waiting for promises is to return the promise ala https://github.com/everydayhero/great-us-crossing/blob/master/source/store/actions/teams/__tests__/fetchTeams-test.js#L52

File: source/Routes/Home/index.js (hunk 1 above)
Comment (bradparker) [inline source/Routes/Home/index.js]: Yep! Cheers.

File: source/store/reducers/charities/__tests__/charities-reducer-test.js [hunk 4]
```diff
@@ -3,21 +3,47 @@ import { expect } from 'chai'
...
+        status: 'fetched',
+        data: [{ slug: 'a-charity' }],
+        error: null
+      })
+    })
+  })
   context('provided action.FETCH_CHARITY', () => {
     it("returns the state with [action.slug] set to { status: 'fetching' }", () => {
-      expect(charities({}, { type: 'FETCH_CHARITY', slug: 'a-charity' })).to.eql({ 'a-charity': { status: 'fetching' } })
+      expect(charities({}, { type: 'FETCH_CHARITY', payload: { slug: 'a-charity' } })).to.eql({ data: [{ slug: 'a-charity', status: 'fetching' }] })
```
Comment (alexander-bobin) [inline source/store/reducers/charities/__tests__/charities-reducer-test.js]: If we only care about `status: 'fetching'` we might be able to shorten this line with `sinon.match.has('status', 'fetching')`

File: source/store/actions/charities/__tests__/fetchCharities-test.js (hunk 2 above)
Comment (bradparker) [inline source/store/actions/charities/__tests__/fetchCharities-test.js]: Oof it is too. It should be a test for the error being present on the action.

File: source/store/actions/charities/__tests__/fetchCharities-test.js (hunk 3 above)
Comment (bradparker) [inline source/store/actions/charities/__tests__/fetchCharities-test.js]: Hm, yeah that is better... presume it works with rejected promises ok?

Comment (alexander-bobin) [inline source/store/actions/charities/__tests__/fetchCharities-test.js]: Pretty sure it will complain instantly for uncaught rejections

**Pull request merged**